src/lib.rs

```diff
@@ -65,15 +65,16 @@ fn map_key_cmp(a: &str, b: &str) -> std::cmp::Ordering {
     }
 }
 
-fn sort_map_keys(keys: &PySequence, len: usize) -> Vec<(&str, usize)> {
+fn sort_map_keys(keys: Bound<PySequence>, len: usize) -> Vec<(String, usize)> {
```

Comment:
You could try using the new `pyo3::pybacked::PyBackedStr` here. It is intended to give the option to avoid the allocation where possible.

Reply:
awesome!! it works like a charm!! thank you! 

a little question about another perf degradation. why decoding of single null/int/bool degraded? 

<details><summary>int</summary>
<p>

im okay of random +perf -perf here but they are all -perf :( 

<img width="855" alt="image" src="https://github.com/MarshalX/python-libipld/assets/15520314/e9660ce8-a38b-40c4-afea-8e4b4c38ced5">

same with null and true/false

</p>
</details>